### PR TITLE
Onboarding: Fix the back button on the Domain screen

### PR DESCRIPTION
**Pull request by arthur791004**
Related to #

## Proposed Changes

* The label of the back button on the Domain screen in the onboarding flow should be `Back` and goes back to the previous step

## Why are these changes being made?

*

## Testing Instructions

* Go to /setup/onboarding
* On the Goals screen, select any goal
* On the Design Picker screen, select any design
* On the Domain screen,
  * Ensure the label of the back button is `Back`
  * Click the back button
  * Ensure you go back to the previous step
* Go to /setup/onboarding?flags=-onboarding/goals-first
* Ensure the label of the back button is `Back to sites`
* Click the back button
* Ensure you go back to /sites

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by okmttdhr — COMMENTED** [review]
Followed the instructions, but the label is "Back to sites" pointing to /sites. 👀 
Screenshot 2024-12-18 at 16 38 43

**Review by candy02058912 — CHANGES_REQUESTED** [review]
I think the changes need to be made in `client/signup/steps/domains/index.jsx` as we're loading unified-domains in /setup/onboarding: https://github.com/Automattic/wp-calypso/blob/216c4cc950d95e9ec8877895cf3086f3986cf8b1/client/landing/stepper/declarative-flow/onboarding.ts#L48

**Comment by arthur791004** [comment]
Yes. I changed the wrong file 🥲

File: client/signup/steps/domains/index.jsx [hunk 1]
```diff
@@ -1349,6 +1350,11 @@ export class RenderDomainsStep extends Component {
 		} else if ( isOnboardingGuidedFlow( flowName ) ) {
 			// Let the framework decide the back url.
 			backUrl = undefined;
+		} else if ( isOnboardingFlow( flowName ) ) {
+			const searchParams = new URLSearchParams( window.location.search );
+			const existingParams = Object.fromEntries( searchParams.entries() );
+			backUrl = getStepUrl( flowName, 'designSetup', null, this.getLocale(), existingParams );
```
Comment (candy02058912) [inline client/signup/steps/domains/index.jsx]: Was unsure how to call the `getBack` navigation from the stepper framework here, so hardcoding the previous step for now cc: @Automattic/vertex

Comment (arthur791004) [reply client/signup/steps/domains/index.jsx]: Passed the `goBack` function by [7b8c0e9](https://github.com/Automattic/wp-calypso/pull/97592/commits/7b8c0e9bca244714fd69d94c54b0edf3b2acd131).

**Review by candy02058912 — APPROVED** [review]
Way better than me trying to hard code the back step. Thanks for the quick update. LGTM!

**Review by p-jackson — APPROVED** [review]
The `/start` and `/setup/onboarding?flags=-onboarding/goals-first` flows still work as before. I'll merge.

For some reason I find I have to click the `Back` button twice 🤔 It seems to have something to do with the input box having focus, because if I remove the focus from the input box first then I only have to click `Back` once.

https://github.com/user-attachments/assets/1e1e12f9-0d92-4acb-8330-c06834b5e39a

**Comment by zaguiini** [comment]
This PR already got merged, but it's not working for me:

image

cc @p-jackson @arthur791004

**Comment by vykes-mac** [comment]
> This PR already got merged, but it's not working for me if I try `trunk` locally:
> 
> image
> 
> cc @p-jackson @arthur791004

It works for me but shows `Back to sites` which I think is being addressed in another PR
image

**Comment by arthur791004** [comment]
Interesting...it works as expected on my env 🤔

**Comment by p-jackson** [comment]
Ooooh! I just managed to reproduce the missing `Back` link when I'm logged out. In this case the domains step comes after the user step. The `Back` link should still go back to the design picker (since the user step should be skipped after you're logged in). But it's missing.

**Comment by arthur791004** [comment]
Not sure what I'm missing but it still works 🙈

**Comment by p-jackson** [comment]
@arthur791004 here's the offending line for me: https://github.com/Automattic/wp-calypso/blob/5a9bd19c6c02908a8cf40ca464896582cd8171d8/client/signup/steps/domains/index.jsx#L1322

I'll spin up a PR. Sadly the logic is about to become more complex 😮‍💨

**Comment by p-jackson** [comment]
Fix is here: https://github.com/Automattic/wp-calypso/pull/97633

**Pull request merged**